### PR TITLE
chore: Rename `develop` to `main`

### PR DESCRIPTION
**Pull request by Gudahtt**
## **Description**

Rename references to the main branch. It will be called `main` rather than `develop`.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28821?quickstart=1)

## **Related issues**

Relates to https://github.com/MetaMask/MetaMask-planning/issues/3677/

## **Manual testing steps**

N/A

## **Screenshots/Recordings**

N/A

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -477,7 +477,7 @@ jobs:
 
   # This job is used for the e2e quality gate.
   # It must be run before any job which uses the run-all.js script.
-  # The job is skipped in develop, master or RC branches.
+  # The job is skipped in main, master or RC branches.
   get-changed-files-with-git-diff:
```
Comment (Gudahtt) [inline .circleci/config.yml:481]: The `git-diff-develop.ts` script should be renamed as well, but we can do that in a later PR

File: .github/scripts/check-template-and-add-labels.ts [hunk 2]
```diff
@@ -334,7 +334,7 @@ function craftRegressionLabel(regressionStage: RegressionStage | undefined, rele
   switch (regressionStage) {
     case RegressionStage.Development:
       return {
-        name: `regression-develop`,
+        name: `regression-main`,
```
Comment (Gudahtt) [inline .github/scripts/check-template-and-add-labels.ts:337]: This label will need to be renamed

File: .circleci/config.yml [hunk 3]
```diff
@@ -250,7 +250,7 @@ workflows:
           filters:
             branches:
               only:
-                - develop
+                - main
```
Comment (Gudahtt) [inline .circleci/config.yml:253]: Seems like this should run on the `master` branch as well, so we could reuse the `main_master_rc_only` alias here. We can do this in a later PR though.

**Review by danjm — APPROVED** [review]
LGTM

**Review by pedronfigueiredo — APPROVED** [review]

**Review by HowardBraham — APPROVED** [review]

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**